app/plugin.go

```diff
@@ -400,6 +402,93 @@ func (a *App) GetPlugins() (*model.PluginsResponse, *model.AppError) {
 	return resp, nil
 }
 
+// GetMarketplacePlugins returns a list of plugins from the marketplace-server,
+// and plugins that are installed locally.
+func (a *App) GetMarketplacePlugins(filter *model.MarketplacePluginFilter) ([]*model.MarketplacePlugin, *model.AppError) {
+	var result []*model.MarketplacePlugin
+	pluginSet := map[string]bool{}
+	pluginsEnvironment := a.GetPluginsEnvironment()
+	if pluginsEnvironment == nil {
+		return nil, model.NewAppError("GetMarketplacePlugins", "app.plugin.config.app_error", nil, "", http.StatusInternalServerError)
+	}
+
+	marketplaceClient, err := marketplace.NewClient(
+		*a.Config().PluginSettings.MarketplaceUrl,
+		a.HTTPService,
+	)
+	if err != nil {
+		return nil, model.NewAppError("GetMarketplacePlugins", "app.plugin.marketplace_client.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	marketplacePlugins, err := marketplaceClient.GetPlugins(filter)
+	if err != nil {
+		return nil, model.NewAppError("GetMarketplacePlugins", "app.plugin.marketplace_plugins.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	for _, plugin := range marketplacePlugins {
+		installedVersion := ""
+		var manifest *model.Manifest
+		if manifest, err = pluginsEnvironment.GetManifest(plugin.Manifest.Id); err == nil {
+			// Plugin is installed.
+			installedVersion = manifest.Version
+		}
+
+		pluginSet[plugin.Manifest.Id] = true
+		result = append(result, &model.MarketplacePlugin{
+			BaseMarketplacePlugin: plugin,
+			InstalledVersion:      installedVersion,
+		})
+	}
+
+	// Include all other installed plugins.
+	plugins, err := pluginsEnvironment.Available()
+	if err != nil {
+		return nil, model.NewAppError("GetMarketplacePlugins", "app.plugin.config.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	for _, plugin := range plugins {
+		if plugin.Manifest == nil || pluginSet[plugin.Manifest.Id] || !pluginMatchesFilter(plugin.Manifest, filter.Filter) {
```

Comment:
Is this only to get the local plugins that are installed but are not in the marketplace? 

The only value I see in this case is that they are able to click `Configure` directly on them but it may be misleading to the admin user if they uploaded an internal plugin thought the System Console and now it's showing in the marketplace list. 🤔

Thoughts @lieut-data, @aaronrothschild ?

Reply:
> Is this only to get the local plugins that are installed but are not in the marketplace?
> 

Yes it is.

> The only value I see in this case is that they are able to click Configure directly in the UI but it may be misleading to the admin user if they uploaded an internal plugin thought the System Console and now it's showing in the marketplace list.

Ah true. But since we're thinking that `Plugins Marketplace` is a one stop shop, then maybe it's ok when they see locally installed plugins there 🤔

Reply:
I think an icon/badge on the plugn to indicate it is "local-only" is the right approach to this.  I do think there will be instances where people expect the marketplace to be the "one-stop-shop" and will be confused if it's NOT there.  

Remind me - wasn't there a panel/tab for "Installed"?

Reply:
Yes, there is: 
- `All` tab that is currently showing all plugins in the marketplace + any internal installed
- `Installed` tab currently showing all installed regardless of origin marketplace or uploaded internally

Reply:
I propose we iterate on any UX improvements independent of this PR. I think we can risk some minor confusion in the MVP.

Reply:
Agreed.

Reply:
I've created [MM-18334](https://mattermost.atlassian.net/browse/MM-18334) to track this for future versions of marketplace.